### PR TITLE
Preparing for resident

### PR DESCRIPTION
**Pull request by aborsu**

### CODE REVIEW

File: kgitb/config.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
+# -*- coding: utf-8 -*-
```
Comment (SagaStalin) [inline kgitb/config.py]: MINOR Missing module docstring [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0111)

File: kgitb/config.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
 from os import environ as env
```
Comment (SagaStalin) [inline kgitb/config.py]: MAJOR Add a docstring to this module. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)

File: kgitb/__main__.py [hunk 3]
```diff
@@ -1,12 +1,17 @@
 import logging
```
Comment (SagaStalin) [inline kgitb/__main__.py]: MAJOR Add a docstring to this module. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
MINOR Missing module docstring [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0111)

File: kgitb/__main__.py [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
 import sys
 
-from .config import GH_TOKEN, HOST, PORT
-from . import app
+from flask import Flask
+
+from .config import GH_TOKEN, HOST, PORT, HIDDEN
+from . import routes
 
 log = logging.getLogger(__name__)
```
Comment (SagaStalin) [inline kgitb/__main__.py]: MINOR Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

File: kgitb/__main__.py [hunk 5]
```diff
@@ -1,12 +1,17 @@
...
 
 
+app = routes.setup(Flask(HIDDEN[1:-1]))
```
Comment (SagaStalin) [inline kgitb/__main__.py]: MINOR Invalid constant name "app" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

File: kgitb/__main__.py [hunk 6]
```diff
@@ -1,12 +1,17 @@
...
+
+
 def main():
```
Comment (SagaStalin) [inline kgitb/__main__.py]: MAJOR Add a docstring to this function. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
MINOR Missing function docstring [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0111)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 102 issues
* MAJOR 50 major
* MINOR 52 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [__init__.py#L1](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/__init__.py#L1): Add a docstring to this module. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. MAJOR [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. MAJOR [github.py#L14](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L14): Add a docstring to this function. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. MAJOR [github.py#L15](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L15): Remove this "\", add another "\" to escape it, or make this a raw string. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1717)
1. MAJOR [github.py#L18](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L18): Add a docstring to this function. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. MAJOR [github.py#L22](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L22): Add a docstring to this function. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. MAJOR [github.py#L27](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L27): Add a docstring to this function. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. MAJOR [github.py#L36](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L36): Add a docstring to this function. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. MAJOR [github.py#L40](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L40): Add a docstring to this function. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. MAJOR [github.py#L59](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L59): Add a docstring to this function. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)

File: kgitb/rules/status_rules.py [hunk 7]
```diff
@@ -4,9 +4,9 @@
 from ..config import TYPES
 
 status_pattern = re.compile(
```
Comment (SagaStalin) [inline kgitb/rules/status_rules.py]: MINOR Invalid constant name "status_pattern" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

File: kgitb/rules/status_rules.py [hunk 8]
```diff
@@ -4,9 +4,9 @@
...
-    """^([^()\s]*)(\s*)\(([^()\s]*)\)(\s*):(\s*)(.*\w)(\s*)$""")
+    r"^([^()\s]*)(\s*)\(([^()\s]*)\)(\s*):(\s*)(.*\w)(\s*)$")
 
-spaces = re.compile("\w*\s+\w*\(|\w+\(.*[^\w].*\):")
+spaces = re.compile(r"\w*\s+\w*\(|\w+\(.*[^\w].*\):")
```
Comment (SagaStalin) [inline kgitb/rules/status_rules.py]: MINOR Invalid constant name "spaces" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

File: kgitb/rules/status_rules.py [hunk 9]
```diff
@@ -4,9 +4,9 @@
...
 
 _type_string = ", ".join(sorted(TYPES))
```
Comment (SagaStalin) [inline kgitb/rules/status_rules.py]: MINOR Invalid constant name "_type_string" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 24 issues
* MAJOR 10 major
* MINOR 14 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. MAJOR [__init__.py#L61](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/__init__.py#L61): Remove the parentheses after this "return" keyword [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1721)
1. MAJOR [__init__.py#L61](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/__init__.py#L61): Remove those useless parentheses [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1110)
1. MAJOR [status_rules.py#L27](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L27): Merge this if statement with the enclosing one. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1066)
1. MAJOR [status_rules.py#L31](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L31): This function has 4 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. MAJOR [status_rules.py#L31](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L31): Function has a complexity of 15 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. MAJOR [status_rules.py#L65](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L65): Remove those useless parentheses [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1110)
1. MAJOR [status_rules.py#L65](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L65): Remove the parentheses after this "if" keyword [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1721)

File: kgitb/__main__.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+def main():
+    app = routes.setup(Flask(HIDDEN[1:-1]))
+    app.run(host=HOST, port=PORT)
+
+
+if __name__ == '__main__':
+    if GH_TOKEN is None:
+        sys.exit("GH_TOKEN must be set.")
+    LOG.basicConfig(level=logging.DEBUG)
```
Comment (ghost) [inline kgitb/__main__.py:20]: MAJOR Instance of 'RootLogger' has no 'basicConfig' member [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE1101)

File: kgitb/rules/__init__.py [hunk 11]
```diff
@@ -38,13 +38,19 @@ def _get_rules(module):
...
 
+def _is_merge(status_line):
+    return status_line.startswith("Merge pull request #")
+
+
 def split_lines(commit_message):
     return re.split('\r?\n', commit_message)
 
 
 def apply(commit_message):
```
Comment (SagaStalin) [inline kgitb/rules/__init__.py]: MINOR Redefining built-in 'apply' [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AW0622)

File: kgitb/rules/__init__.py [hunk 12]
```diff
@@ -38,13 +38,19 @@ def _get_rules(module):
 _status_rules = _get_rules(status_rules)
```
Comment (ghost) [inline kgitb/rules/__init__.py:37]: MINOR Invalid constant name "_status_rules" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

File: kgitb/rules/status_rules.py [hunk 13]
```diff
@@ -23,9 +17,8 @@ def check_status_length(status_line):
...
     'Limit status to 50 characters'
     """
-    if not _is_merge(status_line):
-        if len(status_line) > MAX_STATUS_LENGTH:
-            return "Limit status to {0} characters".format(MAX_STATUS_LENGTH)
+    if len(status_line) > MAX_STATUS_LENGTH:
+        return "Limit status to {0} characters".format(MAX_STATUS_LENGTH)
 
 
 def check_status_formatting(status_line):
```
Comment (ghost) [inline kgitb/rules/status_rules.py:24]: MAJOR Function has a complexity of 13 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 12 issues
* MAJOR 5 major
* MINOR 7 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. MINOR [__init__.py](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/__init__.py): Add a new line at the end of this file "__init__.py". [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS113)
1. MINOR [github.py#L11](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/github.py#L11): Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [routes.py#L6](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/routes.py#L6): Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [__init__.py#L36](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/rules/__init__.py#L36): Invalid constant name "_line_rules" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [__init__.py#L37](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/rules/__init__.py#L37): Invalid constant name "_raw_rules" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

File: setup.py [hunk 14]
```diff
@@ -1,31 +1,25 @@
...
 
 setup(
     name='kgitb',
     packages=['kgitb', 'kgitb.rules'],
     scripts=['bin/resident'],
-    version='0.0.5',
+    version='0.0.6',
     description='A commit message linter',
-    long_description=read_md('README.md'),
+    long_description=io.open(readme, encoding="utf-8").read(),
```
Comment (kxepal) [inline setup.py]: This will cause resource warnings for Python 3 since you didn't close the file object.

File: bin/resident [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+    for file_path in file_paths:
+        with io.open(file_path) as message:
+            errors.append(rules.apply_rules(message.read()))
+    if (len(file_paths) == 1) and (len(errors[0]) != 0):
+        format_errrors(errors[0])
+        return False
+    else:
+        result = True
+        for file_path, file_errors in zip(file_paths, errors):
+            if len(file_errors) != 0:
```
Comment (kxepal) [inline bin/resident]: `if file_errors` is more Pythonic. Same for above `len(errors[0]) != 0`.

File: bin/resident [hunk 16]
```diff
@@ -0,0 +1,202 @@
...
+                format_errrors(file_errors, file_path)
+                result = False
```
Comment (kxepal) [inline bin/resident:183]: If result set to False, may be break out the loop then?

File: setup.py [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+    readme = "README.rst"
+print("Using %s as readme" % readme)
+
+setup(
+    name='kgitb',
+    packages=['kgitb', 'kgitb.rules'],
+    scripts=['bin/resident'],
+    version='0.0.6',
+    description='A commit message linter',
+    long_description=io.open(readme, encoding="utf-8").read(),
```
Comment (kxepal) [inline setup.py]: This will cause resource warnings for Python 3 since you didn't close the file object.

File: bin/resident (hunk 16 above)
Comment (aborsu) [inline bin/resident]: No, the loop copies the old file to a temp file that overwrites the file.
(I couldn't find any better way to modify a line in a file in place).
So if we stop the loop, the new file will be truncated.

This is to play nice with other hooks that might exist.

Comment (aborsu) [reply bin/resident:183]: I thought i would check all the files and comment on all of them.
Avoids having to fix one file and then having the test fail on the following one?

Comment (kxepal) [reply bin/resident:183]: Hm...no, that's a good case.

UPD. I mean, it's good to check all the files and report all the stuff in single shot.

**Comment by SagaStalin** [comment]
SonarQube analysis reported 11 issues
* MAJOR 5 major
* MINOR 6 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. MINOR [__init__.py](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/__init__.py): Add a new line at the end of this file "__init__.py". [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS113)
1. MINOR [github.py#L11](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/github.py#L11): Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [routes.py#L6](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/routes.py#L6): Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [__init__.py#L36](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/rules/__init__.py#L36): Invalid constant name "_line_rules" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [__init__.py#L37](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/rules/__init__.py#L37): Invalid constant name "_raw_rules" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 11 issues
* MAJOR 5 major
* MINOR 6 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. MINOR [__init__.py](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/__init__.py): Add a new line at the end of this file "__init__.py". [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS113)
1. MINOR [github.py#L11](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/github.py#L11): Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [routes.py#L6](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/routes.py#L6): Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [__init__.py#L36](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/rules/__init__.py#L36): Invalid constant name "_line_rules" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. MINOR [__init__.py#L37](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/rules/__init__.py#L37): Invalid constant name "_raw_rules" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

File: kgitb/rules/__init__.py [hunk 18]
```diff
@@ -34,22 +34,25 @@ def _get_rules(module):
             if tup[0][0] != "_"]
 
 _line_rules = _get_rules(line_rules)
```
Comment (ghost) [inline kgitb/rules/__init__.py:36]: MINOR Invalid constant name "_line_rules" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

File: kgitb/rules/__init__.py [hunk 19]
```diff
@@ -34,22 +34,25 @@ def _get_rules(module):
...
     """Apply all rules to the commit message."""
     errors = []
-    for rule in _raw_rules:
-        err = rule(commit_message)
-        if err is not None:
-            errors.append(err)
+    if _is_merge(commit_message):
+        return errors
     commit_lines = split_lines(commit_message)
+    status = commit_lines[0]
```
Comment (ghost) [inline kgitb/rules/__init__.py:54]: MINOR Unused variable 'status' [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AW0612)

**Comment by ghost** [comment]
SonarQube analysis reported 11 issues
- MAJOR 5 major
- MINOR 6 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
2. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
3. MAJOR [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
4. MINOR [**init**.py](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/__init__.py): Add a new line at the end of this file "**init**.py". [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS113)
5. MINOR [github.py#L11](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/github.py#L11): Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
6. MINOR [routes.py#L6](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/routes.py#L6): Invalid constant name "log" [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

**Pull request merged**